crates/type-system/src/ontology/uri/mod.rs

```diff
@@ -87,9 +89,14 @@ impl FromStr for VersionedUri {
     type Err = ParseVersionedUriError;
 
     fn from_str(uri: &str) -> Result<Self, ParseVersionedUriError> {
-        let (base_uri, version) = uri.rsplit_once("/v/").ok_or(ParseVersionedUriError)?;
-
         // TODO: better error handling
+        lazy_static! {
+            static ref RE: Regex = Regex::new(r#"(.*/)v/(\d)*"#).expect("Regex failed to compile");
+        }
```

Comment:
Do we want to use [LazyLock](https://doc.rust-lang.org/nightly/std/sync/struct.LazyLock.html) instead of `lazy_static`?

Reply:
Any preference about these? I'm just following the [regex docs suggestion](https://docs.rs/regex/latest/regex/#example-avoid-compiling-the-same-regex-in-a-loop)

Reply:
I'd probably use `LazyLock`:

```rust
static RE: LazyLock<Regex> = LazyLock::new(|| {
    Regex::new(r#"(.*/)v/(\d)*"#).expect("Regex failed to compile")
});
```

I guess it's not in the regex docs as it's not stabilized yet.